channeldb/migtest/migtest.go

```diff
@@ -2,37 +2,35 @@ package migtest
 
 import (
 	"fmt"
-	"io/ioutil"
 	"os"
 	"testing"
 
 	"github.com/lightningnetwork/lnd/kvdb"
 )
 
-// MakeDB creates a new instance of the ChannelDB for testing purposes. A
-// callback which cleans up the created temporary directories is also returned
-// and intended to be executed after the test completes.
-func MakeDB() (kvdb.Backend, func(), error) {
+// MakeDB creates a new instance of the ChannelDB for testing purposes.
+func MakeDB(t testing.TB) (kvdb.Backend, error) {
 	// Create temporary database for mission control.
-	file, err := ioutil.TempFile("", "*.db")
+	file, err := os.CreateTemp("", "*.db")
 	if err != nil {
-		return nil, nil, err
+		return nil, err
 	}
 
 	dbPath := file.Name()
+	t.Cleanup(func() {
+		file.Close()
+		os.RemoveAll(dbPath)
```

Comment:
And just for completeness' sake, also use `require.NoError()` for the two calls? Here and in other places of this commit?

Reply:
> Since this is a single file, this should probably be `os.Remove(dbPath)`.

`os.Remove(dbPath)` is not sufficient on Windows. Windows does not allow removing an open file (see https://github.com/golang/go/issues/50510#issuecomment-1008535852), so we have to close the open file descriptor explicitly.

Reply:
I just mean `os.Remove(dbPath)` vs. `os.RemoveAll(dbPath)`. I agree with the `file.Close()` call :+1: